File: PCK2018/Q3.cpp
#include <iostream>

int main(){

	int N, C;

	std::cin >> N >> C;

	int sum = 0;

	for(int i = 0; i < C; ++i){

		int input;

		std::cin >> input;

		sum += input;
	}

	if(sum % (N + 1) != 0){
		std::cout << sum / (N + 1) + 1 << std::endl;
	}
	else{
		std::cout << sum / (N + 1) << std::endl;
	}

	return 0;
}

File: Atcoder/ABC117/Q3.py
N, M = map(int, input().split())
X = [int(x) for x in input().split()]

if N >= M:
    print(0)
    quit()

X.sort()

dX = list()

for i in range(M - 1):
    dX.append(X[i + 1] - X[i])

dX.sort(reverse=True)

print(sum(dX) - sum([dX[i] for i in range(N - 1)]))
